controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,7 +106,10 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	if status.Nodes <= *kcp.Spec.Replicas {
+	// RolloutStrategy is currently defaulted and validated to be RollingUpdate
+	// We can ignore MaxUnavailable because we are enforcing health checks before we get here.
+	maxNodes := *kcp.Spec.Replicas + int32(kcp.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntValue())
```

Comment:
@detiber This operation seems to brake testing in here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/4073/pull-cluster-api-test-main/1356238114686767104
Do you have any ideas what are we missing here?